boudicca.base/publisher-event-html/src/main/kotlin/base/boudicca/publisher/event/html/service/EventService.kt

```diff
@@ -83,6 +87,7 @@ class EventService @Autowired constructor(
         return query
     }
 
+    @Suppress("detekt:CyclomaticComplexMethod")
```

Comment:
we should refactor this function. I too think it could be written better

Reply:
how? i think it is long, but it is the same pattern every time, it does not make sense to split it